tests/simulator_test.cpp

```diff
@@ -2,82 +2,266 @@
 #include "main/simulator.h"
 #include "gtest/gtest.h"
 
-// ! OLD ENERGY COLLECTORS TEST
-/*
-TEST_F(SceneManagerTest, EnergyCollectorPositionsCheck) {
-  for (const auto &collectorPtr : manager.getEnergyCollectors()) {
-    std::cout << *collectorPtr << std::endl;
+using constants::kPi;
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+
+const int kSkipNumCollectors = 37;
+const float kSkipFrequency = 1000;
+const Vec3 kVecZero(0, 0, 0);
+const Vec3 kVecUp(0, 0, 1);
+const Vec3 kVecX(1, 0, 0);
+const Vec3 kVecY(0, 1, 0);
+
+float deg2rad(float deg) { return 2 * kPi * deg / 360; }
+
+class FakeModel : public AbstractModel {
+
+public:
+  explicit FakeModel(bool empty) : empty_(empty){};
+  std::vector<objects::TriangleObj *> triangles() const override { return {}; }
+  float height() const override { return 0; }
+  float sideSize() const override { return 0; }
+  bool empty() const { return empty_; }
+
+private:
+  bool empty_;
+};
+
+class EnergyCollectorTest : public ::testing::Test {
+public:
+  EnergyCollectorTest() : nonEmptyModel(false), emptyModel(true){};
+  enum class HitResult { ENERGY_COLLECTORS_EMPTY, NO_HIT, HIT };
```

Comment:
Looks like you learned a new trick and immediately want to use it everywhere. If you used a simple bool code would be much much better (see the side effect comment).